File: shortestWord.py
# Define a function if the shortest word of all other words in the sentence
def shortest_word(sentence):
    # Split the sentence into words
    words = sentence.split()
    
    # If there are no words, return None
    if not words:
        return None
    
    # Find the shortest word using min() function with key=len
    shortest = min(words, key=len)
    
    return shortest

def main():
    sentence = input("Enter a string of words: ")
    shortest = shortest_word(sentence)
    
    if shortest is not None:
        print("The shortest word in the string is:", shortest)
    else:
        print("No words found in the string.")

if __name__ == "__main__":
    main()

File: shortestWord.rs
use std::io;

// Define a function if the shortest word of all other words in the sentence
fn shortest_word(sentence: &str) -> Option<&str> {
    sentence
        .split_whitespace()
        .min_by_key(|word| word.len())
}

fn main() {
    // Input string from the user
    println!("Enter a string of words:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");

    // Find the shortest word in the string
    if let Some(shortest) = shortest_word(&input) {
        println!("Shortest word: {}", shortest);
    } else {
        println!("No words found in the input");
    }
}
